src/models/image-map.ts

```diff
@@ -443,72 +433,34 @@ export const localAssetsImagesHandler: IImageHandler = {
   priority: 2,
 
   match(url: string) {
-    return url ? url.startsWith("assets/") || url.startsWith("curriculum/") : false;
+    return !(url.match(":") || url.match(/^\/dev\/.*?\/portals/));
```

Comment:
OK, I added comments. The pattern with `portals` is for the "original firebase storage path". As I understand it, Firebase storage was changed at some point, but there are possibly instances of the old path format in content.

Would it be overkill to add some functions like the below to use for each handler's `match` method? It might make the local assets image handler's match more understandable. But maybe we don't want the local assets handler to become a default handler?

```
const isExternalUrlImage = (url: string) => /^(https?:\/\/|data:image\/)/.test(url);

const isFirebaseStorageImage = (url: string) => {
  return url.startsWith(kFirebaseStorageUrlPrefix) ||
         /^\/.+\/portals\/.+$/.test(url); // original firebase storage path reference
};

const isFirebaseRealTimeDBImage = (url: string) => {
  return url.startsWith(kFirebaseRTDBFauxUrlPrefix) ||
         (url.startsWith(`${kCCImageScheme}://`) && (url.indexOf("concord.org") < 0));
};

const isLocalAssetsImage = (url: string) => = return !isExternalUrlImage(url) && !isFirebaseStorageImage(url) && !isFirebaseRealTimeDBImage(url);
```

Reply:
Actually, judging by the [currently failing data card spec test](https://github.com/concord-consortium/collaborative-learning/actions/runs/4472303963/jobs/7858293591?pr=1625), we don't want a default handler.

It seems like the current match check in `localAssetsImagesHandler` is already too broad. The data card's CaseAttribute component does a check to see if a value is an image by checking if ImageMap has a matching handler for that value. That's resulting in some values being replaced with a placeholder image.

Reply:
I updated `localAssetsImagesHandler`'s match check in a way that gets that data card spec test to pass. Do [those changes](https://github.com/concord-consortium/collaborative-learning/pull/1625/files#diff-47f8628856140b5d96756e0a6a51728e4149eeaac4ff7ce6c84ac274560f200fR448-R454) seem reasonable?